internal/service/advancedcluster/resource_advanced_cluster_test.go

```diff
@@ -2558,3 +2536,22 @@ func configFCVPinning(orgID, projectName, clusterName string, pinningExpirationD
 		}
 	`, orgID, projectName, clusterName, mongoDBMajorVersion, pinnedFCVAttr)
 }
+
+func checkFCVPinningConfig(mongoDBMajorVersion int, pinningExpirationDate *string, fcvVersion *int) resource.TestCheckFunc {
+	mapChecks := map[string]string{}
+
+	if pinningExpirationDate != nil {
+		mapChecks["pinned_fcv.0.expiration_date"] = *pinningExpirationDate
+	} else {
+		mapChecks["pinned_fcv.#"] = "0"
+	}
+
+	if fcvVersion != nil {
+		mapChecks["pinned_fcv.0.version"] = fmt.Sprintf("%d.0", *fcvVersion)
+	}
+
+	mapChecks["mongo_db_major_version"] = fmt.Sprintf("%d.0", mongoDBMajorVersion)
+	additionalCheck := resource.TestCheckResourceAttrWith(resourceName, "mongo_db_version", acc.MatchesExpression(fmt.Sprintf("%d..*", mongoDBMajorVersion)))
+
+	return acc.CheckRSAndDS(resourceName, admin.PtrString(dataSourceName), admin.PtrString(dataSourcePluralName), []string{}, mapChecks, additionalCheck)
```

Comment:
alternatively you can call local checkAggr to use common adv_cluster checks

Reply:
`CheckRSAndDS` does have the benefit of asserting the plural data source result using the same list and map values provided. I do see checkAggr has specific logic for tpf implementation so might have to adapt to that method when implementing tpf.